spec/acceptance/pyvenv_spec.rb

```diff
@@ -0,0 +1,143 @@
+require 'spec_helper_acceptance'
+
+describe 'python::pyvenv defined resource' do
+  context 'minimal parameters' do
+    # Using puppet_apply as a helper
+    it 'works with no errors' do
+      pp = <<-EOS
+      class { 'python':
+        version => '3',
+        dev     => 'present',
+      }
+      user { 'agent':
+        ensure         => 'present',
+        managehome     => true,
+        home           => '/opt/agent',
+      }
+      group { 'agent':
+        ensure => 'present',
+        system => true,
+      }
+      python::pyvenv { '/opt/agent/venv':
+        ensure     => 'present',
+        systempkgs => true,
+        owner      => 'agent',
+        group      => 'agent',
+        mode       => '0755',
+      }
+      EOS
+
+      # Run it twice and test for idempotency
+      apply_manifest(pp, catch_failures: true)
+      apply_manifest(pp, catch_changes: true)
+    end
+  end
+  context 'with python::pip' do
```

Comment:
I always like to separate contexts visually
```suggestion

  context 'with python::pip' do
```